### PR TITLE
Unattended fix Kibaba config error and alert index

### PR DESCRIPTION
**Pull request by lastranikos**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-packages/issues/1036|

## Description

- Fix to install version development on Debian bullseye64.
- Fixed to install in distributed version.
- Fix Kibana config. The error in certificate setup prevented configuring the Wazuh plugin.
- Fix Wazuh indexes in Elasticsearch.
- Fix Kibana server is not ready yet. In AIO install.

## Tests

**Wazuh manager and Filbeat**
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --development --w12/22/2021 21:02:11 INFO: Installing all necessary utilities for the installation.
12/22/2021 21:02:41 INFO: Done
12/22/2021 21:02:41 INFO: Adding the Wazuh repository.
12/22/2021 21:02:44 INFO: Done
12/22/2021 21:02:44 INFO: Check recommended minimum hardware requirements for Wazuh Manager done.
12/22/2021 21:02:44 INFO: Starting the installation.
12/22/2021 21:02:44 INFO: Installing the Wazuh manager.
12/22/2021 21:05:10 INFO: Done
12/22/2021 21:05:10 INFO: Starting service wazuh-manager...
12/22/2021 21:05:28 INFO: Wazuh-manager started
12/22/2021 21:05:28 INFO: Installing Filebeat.
12/22/2021 21:05:46 INFO: Filebeat installed
12/22/2021 21:05:48 INFO: Done
12/22/2021 21:05:48 INFO: Starting Filebeat.
12/22/2021 21:05:48 INFO: Starting service filebeat...
12/22/2021 21:05:49 INFO: Filebeat started
12/22/2021 21:05:49 INFO: Wazuh installed correctly
12/22/2021 21:05:49 INFO: Setting the Wazuh repository to production
12/22/2021 21:05:49 INFO: Done
12/22/2021 21:05:49 INFO: Installation Finished
|▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇                        | 62%
```
**Wazuh index and Wazuh Dashboard**
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --development --create-certificates --kibana --elasticsearch wazuh-index1
12/22/2021 20:29:49 INFO: Creating the Elasticsearch certificates.
12/22/2021 20:29:49 INFO: Creating Wazuh server certificates.
12/22/2021 20:29:49 INFO: Creating Kibana certificate.
12/22/2021 20:29:49 INFO: Certificates creation finished. They can be found in /tmp/unattended_scripts/cer12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:30:23 INFO: Done
12/22/2021 20:30:23 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:30:23 INFO: Done
12/22/2021 20:30:23 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 20:30:23 INFO: Starting the installation.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:30:23 INFO: Done
12/22/2021 20:30:23 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 20:30:23 INFO: Starting the installation.
12/22/2021 20:30:23 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:30:23 INFO: Done
12/22/2021 20:30:23 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 20:30:23 INFO: Starting the installation.
12/22/2021 20:30:23 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 20:39:22 INFO: Done
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:30:23 INFO: Done
12/22/2021 20:30:23 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 20:30:23 INFO: Starting the installation.
12/22/2021 20:30:23 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 20:39:22 INFO: Done
12/22/2021 20:39:23 INFO: Configuring Elasticsearch.
12/22/2021 20:29:49 INFO: Installing all necessary utilities for the installation.
12/22/2021 20:30:18 INFO: Done
12/22/2021 20:30:18 INFO: Adding the Wazuh repository.
12/22/2021 20:30:23 INFO: Done
12/22/2021 20:30:23 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 20:30:23 INFO: Starting the installation.
12/22/2021 20:30:23 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 20:39:22 INFO: Done
12/22/2021 20:39:23 INFO: Configuring Elasticsearch.
12/22/2021 20:39:24 INFO: Elasticsearch installed.
12/22/2021 20:39:24 INFO: Starting Elasticsearch.
12/22/2021 20:39:24 INFO: Starting service elasticsearch...
12/22/2021 20:39:39 INFO: Elasticsearch started
12/22/2021 20:39:40 INFO: Initializing Elasticsearch.
12/22/2021 20:39:53 INFO: Done
12/22/2021 20:39:54 INFO: Elasticsearch installed correctly
12/22/2021 20:39:54 INFO: Check recommended minimum hardware requirements for Kibana done.
12/22/2021 20:39:54 INFO: Starting the installation.
12/22/2021 20:39:54 INFO: Installing Open Distro for Kibana.
12/22/2021 20:45:25 INFO: Done
12/22/2021 20:45:25 INFO: Configuring Kibana
12/22/2021 20:46:06 INFO: Wazuh Kibana plugin installed.
12/22/2021 20:46:09 INFO: Kibana certificate setup finished.
12/22/2021 20:46:09 INFO: Starting service kibana...
12/22/2021 20:46:10 INFO: Kibana started
12/22/2021 20:46:10 INFO: Initializing Kibana (this may take a while)
12/22/2021 20:46:21 INFO: 
You can access the web interface https://10.10.121.200. The credentials are admin:admin
12/22/2021 20:46:21 INFO: Kibana installed.
12/22/2021 20:46:21 INFO: Kibana installed correctly
12/22/2021 20:46:21 INFO: Setting the Wazuh repository to production
12/22/2021 20:46:21 INFO: Done
12/22/2021 20:46:21 INFO: Installation Finished
|▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇                        | 62%
```

**Test, fix init elastic cluster in AIO:**
```
bash /tmp/unattended_scripts/wazuh_install.sh --all-in-one --local --ignore-health-check --development
12/22/2021 23:26:30 INFO: Creating the Elasticsearch certificates.
12/22/2021 23:26:30 INFO: Creating Wazuh server certificates.
12/22/2021 23:26:30 INFO: Creating Kibana certificate.
12/22/2021 23:26:31 INFO: Certificates creation finished. They can be found in /tmp/unattended_scripts/cer12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:27:05 INFO: Installing the Wazuh manager.
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:27:05 INFO: Installing the Wazuh manager.
12/22/2021 23:29:13 INFO: Done
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:27:05 INFO: Installing the Wazuh manager.
12/22/2021 23:29:13 INFO: Done
12/22/2021 23:29:13 INFO: Starting service wazuh-manager...
12/22/2021 23:29:30 INFO: Wazuh-manager started
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:27:05 INFO: Installing the Wazuh manager.
12/22/2021 23:29:13 INFO: Done
12/22/2021 23:29:13 INFO: Starting service wazuh-manager...
12/22/2021 23:29:30 INFO: Wazuh-manager started
12/22/2021 23:29:30 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:27:05 INFO: Installing the Wazuh manager.
12/22/2021 23:29:13 INFO: Done
12/22/2021 23:29:13 INFO: Starting service wazuh-manager...
12/22/2021 23:29:30 INFO: Wazuh-manager started
12/22/2021 23:29:30 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:38:19 INFO: Done
12/22/2021 23:26:31 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:00 INFO: Done
12/22/2021 23:27:00 INFO: Adding the Wazuh repository.
12/22/2021 23:27:05 INFO: Done
12/22/2021 23:27:05 WARNING: Health-check ignored for AIO.
12/22/2021 23:27:05 INFO: Installing the Wazuh manager.
12/22/2021 23:29:13 INFO: Done
12/22/2021 23:29:13 INFO: Starting service wazuh-manager...
12/22/2021 23:29:30 INFO: Wazuh-manager started
12/22/2021 23:29:30 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:38:19 INFO: Done
12/22/2021 23:38:19 INFO: Configuring Elasticsearch.
12/22/2021 23:38:21 INFO: Elasticsearch installed.
12/22/2021 23:38:21 INFO: Starting Elasticsearch.
12/22/2021 23:38:21 INFO: Starting service elasticsearch...
12/22/2021 23:38:36 INFO: Elasticsearch started
12/22/2021 23:38:37 INFO: Initializing Elasticsearch.
12/22/2021 23:38:50 INFO: Done
12/22/2021 23:38:50 INFO: Installing Filebeat.
12/22/2021 23:39:32 INFO: Filebeat installed
12/22/2021 23:39:34 INFO: Starting service filebeat...
12/22/2021 23:39:35 INFO: Filebeat started
12/22/2021 23:39:35 INFO: Done
12/22/2021 23:39:35 INFO: Installing Open Distro for Kibana.
12/22/2021 23:44:53 INFO: Done
12/22/2021 23:45:29 INFO: Wazuh Kibana plugin installed.
12/22/2021 23:45:29 INFO: Kibana certificate setup finished.
12/22/2021 23:45:31 INFO: Starting service kibana...
12/22/2021 23:45:32 INFO: Kibana started
12/22/2021 23:45:32 INFO: Initializing Kibana (this may take a while)
12/22/2021 23:45:42 INFO: 
You can access the web interface https://<kibana-host-ip>. The credentials are admin:admin
12/22/2021 23:45:42 INFO: Setting the Wazuh repository to production
12/22/2021 23:45:42 INFO: Done
12/22/2021 23:45:43 INFO: Installation Finished
|▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇|100%
```

**Test, fix init elastic cluster in distributed deploy (Wazuh index and Wazuh dashboard):**
```
bash /tmp/unattended_scripts/wazuh_install.sh --local --development --create-certificates --kibana --elasticsearch wazuh-index1
12/22/2021 23:27:54 INFO: Creating the Elasticsearch certificates.
12/22/2021 23:27:54 INFO: Creating Wazuh server certificates.
12/22/2021 23:27:55 INFO: Creating Kibana certificate.
12/22/2021 23:27:55 INFO: Certificates creation finished. They can be found in /tmp/unattended_scripts/cer12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:29:05 INFO: Done
12/22/2021 23:29:05 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:29:05 INFO: Done
12/22/2021 23:29:05 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 23:29:05 INFO: Starting the installation.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:29:05 INFO: Done
12/22/2021 23:29:05 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 23:29:05 INFO: Starting the installation.
12/22/2021 23:29:05 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:29:05 INFO: Done
12/22/2021 23:29:05 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 23:29:05 INFO: Starting the installation.
12/22/2021 23:29:05 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:38:11 INFO: Done
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:29:05 INFO: Done
12/22/2021 23:29:05 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 23:29:05 INFO: Starting the installation.
12/22/2021 23:29:05 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:38:11 INFO: Done
12/22/2021 23:38:11 INFO: Configuring Elasticsearch.
12/22/2021 23:27:55 INFO: Installing all necessary utilities for the installation.
12/22/2021 23:29:01 INFO: Done
12/22/2021 23:29:01 INFO: Adding the Wazuh repository.
12/22/2021 23:29:05 INFO: Done
12/22/2021 23:29:05 INFO: Check recommended minimum hardware requirements for Elasticsearch done.
12/22/2021 23:29:05 INFO: Starting the installation.
12/22/2021 23:29:05 INFO: Installing Open Distro for Elasticsearch.
12/22/2021 23:38:11 INFO: Done
12/22/2021 23:38:11 INFO: Configuring Elasticsearch.
12/22/2021 23:38:13 INFO: Elasticsearch installed.
12/22/2021 23:38:14 INFO: Starting Elasticsearch.
12/22/2021 23:38:14 INFO: Starting service elasticsearch...
12/22/2021 23:38:30 INFO: Elasticsearch started
12/22/2021 23:38:30 INFO: Initializing Elasticsearch.
12/22/2021 23:38:42 INFO: Done
12/22/2021 23:38:43 INFO: Elasticsearch installed correctly
12/22/2021 23:38:43 INFO: Check recommended minimum hardware requirements for Kibana done.
12/22/2021 23:38:43 INFO: Starting the installation.
12/22/2021 23:38:43 INFO: Installing Open Distro for Kibana.
12/22/2021 23:44:18 INFO: Done
12/22/2021 23:44:18 INFO: Configuring Kibana
12/22/2021 23:44:57 INFO: Wazuh Kibana plugin installed.
12/22/2021 23:45:01 INFO: Kibana certificate setup finished.
12/22/2021 23:45:01 INFO: Starting service kibana...
12/22/2021 23:45:02 INFO: Kibana started
12/22/2021 23:45:02 INFO: Initializing Kibana (this may take a while)
12/22/2021 23:45:12 INFO: 
You can access the web interface https://10.10.121.200. The credentials are admin:admin
12/22/2021 23:45:12 INFO: Kibana installed.
12/22/2021 23:45:12 INFO: Kibana installed correctly
12/22/2021 23:45:13 INFO: Setting the Wazuh repository to production
12/22/2021 23:45:13 INFO: Done
12/22/2021 23:45:13 INFO: Installation Finished
|▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇                        | 62%
```

### CODE REVIEW

**Review by verdx — APPROVED** [review]
LGTM

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes.

File: unattended_scripts/install_functions/opendistro/wazuh.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 installWazuh() {
     
-    logger "Installing the Wazuh manager."
+    logger "Installing the Wazuh Manager."
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/wazuh.sh]: Done!

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 2]
```diff
@@ -8,11 +8,15 @@
 repogpg="https://packages.wazuh.com/key/GPG-KEY-WAZUH"
 repobaseurl="https://packages.wazuh.com/4.x"
 reporelease="stable"
+filebeat_wazuh_template="https://raw.githubusercontent.com/wazuh/wazuh/4.2/extensions/elasticsearch/7.x/wazuh-template.json"
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh]: Done!

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 3]
```diff
@@ -8,11 +8,15 @@
...
+filebeat_wazuh_module="'${repobaseurl}'/filebeat/wazuh-filebeat-0.1.tar.gz"
+kibana_wazuh_module="'${repobaseurl}'/ui/kibana/wazuh_kibana-'${wazuh_version}'_'${elasticsearch_oss_version}'-'${wazuh_kibana_plugin_revision}'.zip"
 
 if [ -n "${development}" ]; then
     repogpg="https://packages-dev.wazuh.com/key/GPG-KEY-WAZUH"
     repobaseurl="https://packages-dev.wazuh.com/pre-release"
     reporelease="unstable"
+    filebeat_wazuh_template="https://raw.githubusercontent.com/wazuh/wazuh/4.3/extensions/elasticsearch/7.x/wazuh-template.json"
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh]: Done!

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 4]
```diff
@@ -72,7 +76,7 @@ installPrerequisites() {
         eval "zypper -n install libcap-progs tar ${debug} || zypper -n install libcap2 tar ${debug}"
     elif [ ${sys_type} == "apt-get" ]; then
         eval "apt-get update -q $debug"
-        eval "apt-get install apt-transport-https curl unzip wget libcap2-bin tar -y ${debug}"
+        eval "apt-get install apt-transport-https curl unzip wget libcap2-bin tar gnupg gnupg2 gnupg1 -y ${debug}"
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh]: Done!

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 5]
```diff
@@ -8,11 +8,17 @@
...
+filebeat_wazuh_module="'${repobaseurl}'/filebeat/wazuh-filebeat-0.1.tar.gz"
+kibana_wazuh_module="'${repobaseurl}'/ui/kibana/wazuh_kibana-'${wazuh_version}'_'${elasticsearch_oss_version}'-'${wazuh_kibana_plugin_revision}'.zip"
 
 if [ -n "${development}" ]; then
     repogpg="https://packages-dev.wazuh.com/key/GPG-KEY-WAZUH"
     repobaseurl="https://packages-dev.wazuh.com/pre-release"
     reporelease="unstable"
+    filebeat_wazuh_template="https://raw.githubusercontent.com/wazuh/wazuh/'${wazuh_major}'/extensions/elasticsearch/7.x/wazuh-template.json"
+    filebeat_wazuh_module="'${repobaseurl}'/filebeat/wazuh-filebeat-0.1.tar.gz"
+    kibana_wazuh_module="'${repobaseurl}'/ui/kibana/wazuh_kibana-'${wazuh_version}'_'${elasticsearch_oss_version}'-'${wazuh_kibana_plugin_revision}'.zip"
```
Comment (rauldpm) [inline unattended_scripts/install_functions/opendistro/common.sh:21]: This lines are identical to 11, 12 and 13, why did you add them here?

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, review this change please

File: unattended_scripts/install_functions/opendistro/common.sh (hunk 5 above)
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh:21]: Change the value of repobaseurl. Now that I see line 19, it could be withdrawn.

Comment (rauldpm) [reply unattended_scripts/install_functions/opendistro/common.sh:21]: Initialize them after if statement, avoid duplicate code

Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh]: Done!

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: unattended_scripts/config/opendistro/certificate/config.yml [hunk 6]
```diff
@@ -11,10 +11,10 @@ nodes:
   wazuh_servers:
     name: <wazuh-node-name>
     ip: <wazuh-manager-ip>
-    # node_type: master
+    # node_type: master # only use in multi-node installation
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/config/opendistro/certificate/config.yml]: Done!

File: unattended_scripts/config/opendistro/certificate/config.yml [hunk 7]
```diff
@@ -11,10 +11,10 @@ nodes:
...
     # name: <wazuh-node-name>
     # ip: <wazuh-manager-ip>
-    # node_type: worker
+    # node_type: worker # only use in multi-node installation
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/config/opendistro/certificate/config.yml]: Done!

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 8]
```diff
@@ -73,7 +77,7 @@ installPrerequisites() {
         eval "zypper -n install libcap-progs tar ${debug} || zypper -n install libcap2 tar ${debug}"
     elif [ ${sys_type} == "apt-get" ]; then
         eval "apt-get update -q $debug"
-        eval "apt-get install apt-transport-https curl unzip wget libcap2-bin tar -y ${debug}"
+        eval "apt-get install apt-transport-https curl unzip wget libcap2-bin tar gnupg -y ${debug}"
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh:80]: Done

File: unattended_scripts/install_functions/opendistro/elasticsearch.sh [hunk 9]
```diff
@@ -180,7 +180,8 @@ initializeElasticsearch() {
 
     if [ ${#elasticsearch_node_names[@]} -eq 1 ]; then
         eval "export JAVA_HOME=/usr/share/elasticsearch/jdk/"
-        eval "/usr/share/elasticsearch/plugins/opendistro_security/tools/securityadmin.sh -cd /usr/share/elasticsearch/plugins/opendistro_security/securityconfig/ -nhnv -cacert /etc/elasticsearch/certs/root-ca.pem -cert /etc/elasticsearch/certs/admin.pem -key /etc/elasticsearch/certs/admin-key.pem -h ${elasticsearch_node_ips[pos]} ${debug}"
+        eval "/usr/share/elasticsearch/plugins/opendistro_security/tools/securityadmin.sh -cd /usr/share/elasticsearch/plugins/opendistro_security/securityconfig/ -icl -nhnv -cacert /etc/elasticsearch/certs/root-ca.pem -cert /etc/elasticsearch/certs/admin.pem -key /etc/elasticsearch/certs/admin-key.pem -h ${elasticsearch_node_ips[pos]} ${debug}"
+        eval "curl ${filebeat_wazuh_template} | curl -X PUT "https://${elasticsearch_node_ips[pos]}:9200/_template/wazuh" -H 'Content-Type: application/json' -d @- -uadmin:admin -k ${debug}"
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/elasticsearch.sh:184]: Done

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 10]
```diff
@@ -5,16 +5,20 @@
...
     reporelease="unstable"
+else
+    repogpg="https://packages.wazuh.com/key/GPG-KEY-WAZUH"
+    repobaseurl="https://packages.wazuh.com/4.x"
+    reporelease="stable"
 fi
 
+filebeat_wazuh_template="https://raw.githubusercontent.com/wazuh/wazuh/'${wazuh_major}'/extensions/elasticsearch/7.x/wazuh-template.json"
+filebeat_wazuh_module="'${repobaseurl}'/filebeat/wazuh-filebeat-0.1.tar.gz"
+kibana_wazuh_module="'${repobaseurl}'/ui/kibana/wazuh_kibana-'${wazuh_version}'_'${elasticsearch_oss_version}'-'${wazuh_kibana_plugin_revision}'.zip"
```
[the remark this answers is not in the document]
Comment (lastranikos) [reply unattended_scripts/install_functions/opendistro/common.sh]: Done!

File: unattended_scripts/install_functions/opendistro/common.sh [hunk 11]
```diff
@@ -67,13 +71,13 @@ installPrerequisites() {
...
-        eval "yum install curl unzip wget libcap tar -y ${debug}"
+        eval "yum install curl unzip wget libcap tar gnupg openssl -y ${debug}"
     elif [ ${sys_type} == "zypper" ]; then
         eval "zypper -n install curl unzip wget ${debug}"         
-        eval "zypper -n install libcap-progs tar ${debug} || zypper -n install libcap2 tar ${debug}"
+        eval "zypper -n install libcap-progs tar gnupg openssl ${debug} || zypper -n install libcap2 tar gnupg openssl ${debug}"
     elif [ ${sys_type} == "apt-get" ]; then
         eval "apt-get update -q $debug"
-        eval "apt-get install apt-transport-https curl unzip wget libcap2-bin tar -y ${debug}"
+        eval "apt-get install apt-transport-https curl unzip wget libcap2-bin tar gnupg openssl -y ${debug}"
```
Comment (rauldpm) [inline unattended_scripts/install_functions/opendistro/common.sh:80]: In wazuh-cert-tool.sh file:

- Add a method to check if openssl is installed using `command -v openssl`, this method should return 1 (installed) or 0 (not installed).
- Add a method in the main that checks the return value of the previous method
  - Returned value is 1: continue
  - Returned value is 0: error message and exit

In common.sh file:

- Before install the prerequisites, call the wazuh-cert-tool.sh file method to check if openssl is installed, if not, add to variable the string: `openssl`
  - openssl="openssl"
- Add this new var to install command, if its empty, it will not install openssl, if has the openssl value, it will install openssl
  - .... install .... ${openssl} -y ...

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, minor change, check if openssl is installed or not, if not, install it.

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

**Review by alberpilot — APPROVED** [review]
LGTM

File: unattended_scripts/install_functions/opendistro/common.sh (hunk 11 above)
Comment (alberpilot) [reply unattended_scripts/install_functions/opendistro/common.sh]: Something like this was implemented.
